src/Form/Input.js

```diff
@@ -271,12 +325,18 @@ class Input extends Component {
         <InputContainer styles={rest.styles}>
           {floating && Label}
 
-          {multiline ? <StyledTextArea {...inputProps} /> : <StyledInput {...inputProps} />}
+          {icon && <StyledIcon size={iconSize} color={iconColor} name={icon} />}
+
+          {multiline ? <StyledTextArea {...inputProps} /> : <StyledInput hasIcon={!!icon} {...inputProps} />}
+
+          {isClearable && stateValue && (
+            <CloseIcon name="close-circle" color="greyDarker" size={18} onClick={this.handleClear} />
```

Comment:
What do you think about making this more configurable? We have a small eye on our mobile app that toggles password visibility - this may be a good way to do this extensibly on the web. The tradeoff will add a little more boilerplate code for clearability.

```jsx
<Input
  leftIcon="security-lock"
  leftIconSize={10}
  leftIconColor={14}
  onLeftIconClick={}
  rightIcon="eye-closed"
  rightIconSize
  rightIconColor
  onRightIconClick={}
/>
```

Reply:
I like it. We are also adding the eye to the web app as well, which will be helpful there too. 👍